src/tlo/methods/cervical_cancer.py

```diff
@@ -0,0 +1,1802 @@
+        df.at[child_id, "ce_ever_diagnosed"] = False
+        df.at[child_id, "ce_cured_date_cc"] = pd.NaT
+        df.at[child_id, "ce_date_last_screened"] = pd.NaT
+
+    def report_daly_values(self):
+
+        # This must send back a dataframe that reports on the HealthStates for all individuals over the past month
+
+        df = self.sim.population.props  # shortcut to population properties dataframe for alive persons
+
+        disability_series_for_alive_persons = pd.Series(index=df.index[df.is_alive], data=0.0)
+
+        # Assign daly_wt to those with cancer stages before stage4 and have either never been treated or are no longer
+        # in the stage in which they were treated
+        disability_series_for_alive_persons.loc[
+            (
+                (df.ce_hpv_cc_status == "stage1") |
+                (df.ce_hpv_cc_status == "stage2a") |
+                (df.ce_hpv_cc_status == "stage2b") |
+                (df.ce_hpv_cc_status == "stage3")
+            )
+        ] = self.daly_wts['stage_1_3']
+
+        # Assign daly_wt to those with cancer stages before stage4 and who have been treated and who are still in the
+        # stage in which they were treated.
+        disability_series_for_alive_persons.loc[
+            (
+                ~pd.isnull(df.ce_date_treatment) & (
+                    (df.ce_hpv_cc_status == "stage1") |
+                    (df.ce_hpv_cc_status == "stage2a") |
+                    (df.ce_hpv_cc_status == "stage2b") |
+                    (df.ce_hpv_cc_status == "stage3")
+                ) & (df.ce_hpv_cc_status == df.ce_stage_at_which_treatment_given)
+            )
+        ] = self.daly_wts['stage_1_3_treated']
+
+        # Assign daly_wt to those in stage4 cancer (who have not had palliative care)
+        disability_series_for_alive_persons.loc[
+            (df.ce_hpv_cc_status == "stage4") &
+            (pd.isnull(df.ce_date_palliative_care))
+            ] = self.daly_wts['stage4']
+
+        # Assign daly_wt to those in stage4 cancer, who have had palliative care
+        disability_series_for_alive_persons.loc[
+            (df.ce_hpv_cc_status == "stage4") &
+            (~pd.isnull(df.ce_date_palliative_care))
+            ] = self.daly_wts['stage4_palliative_care']
+
+        return disability_series_for_alive_persons
+
+    def do_at_generic_first_appt(
+        self,
+        person_id: int,
+        individual_properties: IndividualProperties,
+        symptoms: List[str],
+        schedule_hsi_event: HSIEventScheduler,
+        **kwargs,
+    ) -> None:
+        if 'vaginal_bleeding' in symptoms:
+            schedule_hsi_event(
+                HSI_CervicalCancerPresentationVaginalBleeding(
+                    person_id=person_id,
+                    module=self
+                ),
+                priority=0,
+                topen=self.sim.date,
+                tclose=None)
+
+    def perform_cin_procedure(self, hsi_event, person_id):
```

Comment:
is name right? I think it does not perform the procedue, but decide which one to perform and then schedule the HSI that does the performance, right? So it sould be 'decide_which_cin_removal_procedure_and_schedule()'